addons/purchase/views/purchase_views.xml

```diff
@@ -527,6 +527,25 @@
                 </xpath>
             </field>
         </record>
+
+        <record model="ir.actions.act_window" id="action_open_purchase_from_procurement">
+            <field name="name">action.open.purchase.from.procurement</field>
+            <field name="domain">[('id','=',context.get('active_id'))]</field>
```

Comment:
What's the point? Why not just provide a form opening on that specific record? Also the record is already provided by res_id being set in the method hook.